### PR TITLE
Allow the user to pass down extra environment from the config file.

### PR DESCRIPTION
**Pull request by vext01**
Needed for forcing our GCC's libs into the LD_LIBRARY_PATH.

A PR for the warmup experiment config file will follow this PR.

OK?

### CODE REVIEW

File: krun/tests/env.krun [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import os
+from krun.vm_defs import PythonVMDef, PyPyVMDef
+from krun import EntryPoint
+
+MAIL_TO = []
+
+HEAP_LIMIT = 2097152  # K == 2Gb
```
Comment (ltratt) [inline krun/tests/env.krun]: Is this actually kilobytes or should it just be bytes?

**Comment by ltratt** [comment]
One minor comment; otherwise looks OK (though I was surprised users can't override Krun's environment, and can only append to it).

File: krun/tests/env.krun (hunk 1 above)
Comment (vext01) [inline krun/tests/env.krun]: It's passed to ulimit, which measures in kb.

Comment (ltratt) [inline krun/tests/env.krun]: OK, make sense. Can we use "KiB" not "K", please?

**Comment by vext01** [comment]
Until now we have never needed to allow the config file to adjust the environment. Regarding append: krun uses some environment variables internally to make the benchmarks run (e.g. LD_LIBRARY_PATH for PyPy, and CLASSPATH for Java). The user shouldn't be able to overwrite those, otherwise the benchmarks won't run.

**Comment by ltratt** [comment]
OK, then we should probably throw an error if the user specifies those, otherwise they're going to be deeply confused that they're not having an effect?

**Comment by vext01** [comment]
We shouldn't throw an error. It should be possible for the user to append to these variables, but not overwrite them. See `krun/env.py`.

**Comment by vext01** [comment]
To be clear, the interface I just implemented only ever appends the user's environment to whatever Krun needs to do.

File: krun/vm_defs.py [hunk 2]
```diff
@@ -70,6 +70,16 @@ def __init__(self, iterations_runner):
         # environment changes specific to a benchmark.
         self.common_env_changes = []
 
+        # The user can pass in a dict to influence the environment.
+        # These variables are *appended* (they do not override Krun).
```
Comment (ltratt) [inline krun/vm_defs.py]: I see the confusion: what this comment means is that if someone specifies (e.g.) `LD_LIBRARY_PATH` then the users values are appended to those set by krun. Let's reword the comment a little bit to make that clearer.

**Comment by vext01** [comment]
I've removed the 'K == 2GB' comments from tests, leaving the revised comment only in the example krun config file. I also revised the confusing comment.

Now let me see why tests failed...

**Comment by vext01** [comment]
OK, try this Laurie.

**Comment by vext01** [comment]
(Please allow time for squashing)

**Comment by vext01** [comment]
I think this is a bit overkill, but here's the change. OK?

**Comment by ltratt** [comment]
Works for me (given that I didn't understand things at first, I really don't think it's overkill!).

Squash and then I'll merge?

**Comment by vext01** [comment]
I suppose it's easy to understand code that you wrote yourself and that you have read hundreds of times ;)

(Splat)

**Pull request merged**